### PR TITLE
ANML Network and Dendritic Network Configs

### PR DESCRIPTION
**Pull request by mvacaporale**
This updates the `dendrites.py` config given the latest changes to the framework and introduces configs to replicate ANML in our repo.

Dendrites (`metacl_dendrites`):
|   Num Classes | Meta-test test   | Meta-test train   |    LR   |
|--------------:|:-----------------|:------------------|--------:|
|            10 | 0.20 ± 0.08      | 0.21 ± 0.07       | 0.001   |
|            50 | 0.04 ± 0.01      | 0.04 ± 0.01       | 0.001   |
|           100 | 0.03 ± 0.01      | 0.03 ± 0.01       | 0.001   |
|           200 | 0.01 ± 0.01      | 0.01 ± 0.01       | 0.001   |
|           600 | 0.00 ± 0.00      | 0.01 ± 0.00       | 0.001   |

Vernon Replication (`metacl_anml_replicate`) vs ANML's Repository
image

I'm not sure why ANML performs poorly in our repo. It deserves further investigation, but I will focus on the dendrites network more for now.

### CODE REVIEW

File: projects/meta_cl/experiments/anml_replicate.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        img = pil_image.convert("L").convert("RGB")
+        img = img.resize((28, 28), resample=Image.LANCZOS)
+        img = np.array(img, dtype=np.float32)
+        img = self.to_tensor(img)
+        return self.normalize(img)
+
+
+# ANML-repo results:
+# |-----------------------------------------------------------|
+# | Num Classes | Meta test-test   | Meta test-train  |  LR   |
```
Comment (lucasosouza) [inline projects/meta_cl/experiments/anml_replicate.py:61]: This table is empty, should it be here?

File: projects/meta_cl/networks/anml_networks.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    def forward(self, x):
+        pred = self.prediction(x)
+        mod = self.neuromodulation(x)
+        out = pred * mod
+        out = self.classifier(out)
+        return out
+
+    def eval(self):  # noqa
+        """The ANML repo keeps its model always in training mode."""
```
Comment (lucasosouza) [inline projects/meta_cl/networks/anml_networks.py:92]: Curious. Besides impacting batch norm (eval use calculated stats) and dropout (all neurons active on eval), which this network doesn't use, I don't see any other difference resulting from setting the model to train or eval mode.

**Review by lucasosouza — APPROVED** [review]
I will compare against ANML repo and try to spot the differences after it's merged.

File: projects/meta_cl/experiments/anml_replicate.py (hunk 1 above)
Comment (mvacaporale) [reply projects/meta_cl/experiments/anml_replicate.py]: Good point, fix it. I meant to fill in the table after it was finished running.

File: projects/meta_cl/networks/anml_networks.py (hunk 2 above)
Comment (mvacaporale) [reply projects/meta_cl/networks/anml_networks.py:92]: Correct, that's the intention. I overwrote the `eval` method as our `evaluate_model` function (from `model_utils.py`) calls `model.eval()`.

**Comment by mvacaporale** [comment]
@karangrewal I'm going to merge this so Lucas is unblocked in doing his own comparison of our repo vs ANML's. Please feel free to give feedback still if you like and we can incorporate anything after the fact.
